src/framework/accessibility/internal/accessibilityconfiguration.h

```diff
@@ -28,12 +28,14 @@
 #include "ui/inavigationcontroller.h"
 
 namespace muse::accessibility {
-class AccessibilityConfiguration : public IAccessibilityConfiguration
+class AccessibilityConfiguration : public IAccessibilityConfiguration, public Injectable
 {
-    Inject<ui::INavigationController> navigationController;
+    Inject<ui::INavigationController> navigationController { this };
 
 public:
-    AccessibilityConfiguration() = default;
+    AccessibilityConfiguration(const modularity::ContextPtr& ctx)
+        : Injectable(ctx) {}
```

Comment:
I love this!

It does seem to require that the context field (from the `Injectable` base class) is initialised _before_ the `navigationController` field is initialised with `this`, but I guess that works correctly because members of base classes are _always_ initialised first, right?

Reply:
Right!

```
class Injectable
{
public:
    Injectable(int id)
        : m_id(id)
    {
        qDebug() << "Injectable()";
    }

    int id() const { return m_id; }

private:

    int m_id = 0;
};

struct Inject
{
    Inject(const Injectable* key)
        : m_key(key)
    {
        qDebug() << "Inject()";
    }

    void get() const
    {
        qDebug() << "m_key: " << m_key->id();
    }

private:
    const Injectable* m_key = nullptr;
};

class SomeObject : public Injectable
{
    Inject inj1 { this };

public:

    SomeObject(int id)
        : Injectable(id)
    {
        qDebug() << "SomeObject()";
    }

    void doit()
    {
        inj1.get();
    }
};

int main(int argc, char* argv[])
{
    SomeObject o1(42);
    o1.doit();
}
```

output:
```
Injectable()
Inject()
SomeObject()
m_key:  42
```